docs/user-guide.rst

```diff
@@ -104,9 +104,12 @@ Building TF-A
        export CROSS_COMPILE=<path-to-aarch32-gcc>/bin/arm-linux-gnueabihf-
 
    It is possible to build TF-A using clang or Arm Compiler 6. To do so
-   ``CC`` needs to point to the clang or armclang binary. Only the compiler
-   is switched; the assembler and linker need to be provided by the GNU
-   toolchain, thus ``CROSS_COMPILE`` should be set as described above.
+   ``CC`` needs to point to the clang or armclang binary, but be aware
+   that the GNU linker is used by default. In case of being needed the
```

Comment:
minor: It would be good to state that when CC is changed, the clang assembler (as well as the compiler) is used.